lib/src/analyzer_plugin/analyzer_plugin.dart

```diff
@@ -174,13 +174,18 @@ class MetricsAnalyzerPlugin extends ServerPlugin {
     ResolvedUnitResult analysisResult,
   ) {
     try {
-      if (analysisResult.unit != null) {
+      final path = analysisResult.path;
+      if (analysisResult.unit != null &&
+          path != null &&
+          (driver.analysisContext?.contextRoot.isAnalyzed(path) ?? false)) {
```

Comment:
Do we really need to check this? Seems that `analysisResult.unit` covers this case.

Reply:
And why not to check it for cli?